### PR TITLE
Auto-convert Joi errors to Boom 400

### PR DESCRIPTION
**Pull request by mgreystone**
Auto-convert thrown Joi errors to 400 Bad Request responses. Matches Paperplane's behavior. Keeps us from having lots of `validate(schema, payload).catch(throwBadRequest)` statements & will allow us keep the service files a little more composable.

### CODE REVIEW

File: server/fromError.js [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
+const { converge, identity, path, pipe, prop, unless, when } = require('ramda')
+
+const { boomify } = Boom
 
 const formatError = ({ data, message, error: name, statusCode: status }) =>
   ({ data, message, name, status })
 
+const joiError = pipe(
+  prop('details'),
+  converge(Boom.badRequest, [ prop('message'), identity ])
```
Comment (flintinatux) [inline server/fromError.js]: Let's destructure `badRequest` off of `Boom` like `boomify` is, just for consistency.

**Review by flintinatux — COMMENTED** [review]
I like the code.  Will be nice to get 400's back instead of 500's.  Just one comment.

**Comment by flintinatux** [comment]
joi to the squirrel

**Comment by rpearce** [comment]
@mgreystone reckon we'll be able to get rid of these lines now? If so, I'll take care of it:
https://github.com/articulate/rise-api/blob/master/server/sockets/courses/setLive.js#L36

File: package.json [hunk 2]
```diff
@@ -42,6 +42,7 @@
     "chai": "^4.1.2",
     "coveralls": "^3.0.0",
     "eslint": "^4.11.0",
+    "joi": "12",
```
Comment (rpearce) [inline package.json:45]: bold semver number

Comment (mgreystone) [reply package.json:45]: Oh, boo... Yarn fail.

**Pull request merged**